pkg_resources/__init__.py

```diff
@@ -2016,8 +2016,17 @@ def find_on_path(importer, path_item, only=False):
                 )
             )
         else:
+            # first filter entries that interest us before sorting
+            entries = (
+                entry
+                for entry in os.listdir(path_item)
+                if (entry.lower().endswith('.egg-info')
```

Comment:
Would it be possible to apply this test as an argument to `filter`, such that `entries = filter(of_interest, os.listdir(path_item))`? That function `of_interest` could then have a docstring describing its purpose. My only worry is that the additional n function calls might obviate the performance advantages.

Reply:
Considering how long is the parsing of version (50% cost of exec time IIRC), I don't think it can be worse. I'll change that.